### ISSUE TITLE
contentabc.com

### ISSUE THREAD

**Issue by SampeiNihira** [uBlockOrigin/uBOL-home#214]
There is a blocking rule in the EasyList filter list for the website specified in the title.
It is possible to see that with uBlock Origin the following is applied:

![0](https://github.com/user-attachments/assets/331afcf5-226c-4e0e-b315-733d971e7acd)

Instead it seems to me that with uBlock Origin Lite this blocking rule is not applied.
Is it possible to know the reason?
Thank you.

**Comment by gorhill** [comment]
It is applied?

![image](https://github.com/user-attachments/assets/620eed79-2150-4594-aef5-9762613e759d)

If your question is really why there is no nice informative blocking page, it's because of MV3 limitations.

uBO uses a heuristic to choose whether to block or not blocked a top document, and when it blocked it redirect to a dynamically crafted extension URL, something which can't be directly ported to MV3.

Whether it is possible to craft a set of DNR rules to enforce block-and-redirect top documents is something still left to explore.

**Comment by SampeiNihira** [comment]
I didn't understand what the block page looks like to the user.
I am obliged to give another example.
The website below has a blocking rule in both EasyList (uBlock Origin Lite) but also in AG Base (AG Ablocker v.5.0.113).

![12](https://github.com/user-attachments/assets/fbd5aa85-119d-437b-826b-f6007c2db064)

![13](https://github.com/user-attachments/assets/bf09645d-b96d-40e6-bd1b-2bf053bce8d2)

It seems to me that there is a significant difference........

**Comment by stephenhawk8054** [comment]
> It is applied?

I don't think it's applied. The message when a site is blocked should be `ERR_BLOCKED_BY_CLIENT`.

I checked another site `1nimo.com` which should be blocked too, but I don't see it being blocked in uBOL either.

**Comment by gorhill** [comment]
> I don't think it's applied. The message when a site is blocked should be `ERR_BLOCKED_BY_CLIENT`.

Ah ok, so it's a case of the site not existing. Looking at the [documentation](https://developer.chrome.com/docs/extensions/reference/api/declarativeNetRequest#type-QueryTransform), it says:

> If [no resource types] is specified, all resource types except "main_frame" are blocked

So from memory I left out strict-blocking in uBOL since it's based on a heuristic in uBO. So I come back to my previous answer, something still left to explore.

**Comment by stephenhawk8054** [comment]
Yeah, I remember strict-blocking in uBOL needs to be specified with `doc` somewhere, but can't remember which discussion.

**Comment by krystian3w** [comment]
> Ah ok, so it's a case of the site not existing. 

This domain exists mainly as subdomains:

```
ads2.contentabc.com - 404 HTML
ads2.contentabc.com/ads?spot_id=4563949 - 50x HTML
vz-cdn.contentabc.com - 50x HTML
```

I don't know if it has active advertising/tracking campaigns without HTML 404 and 50x errors.

**Comment by garry-ut99** [comment]
> stephenhawk8054

Yes, `$doc` (or `$document`) works in uBOL, for example `123-stream.org` is strict-blocked by `||123-stream.org^$document` from EasyList, while `||randomadsrv.com^` or `||1nimo.com^` from EasyList is not.

**Comment by SampeiNihira** [comment]
I did a test with the 3 websites.
All three are blocked at the DNS level (I used NextDNS) by OISD + HaGeZi Multi Pro++.
Strict extension blocking would take priority over DNS level intervention.
These 2 filter lists make up for uBOL's lack of strict blocking.

![1](https://github.com/user-attachments/assets/8a693084-8b70-41ca-a814-a35fa40a0523)
![2](https://github.com/user-attachments/assets/592f2206-e0ef-4725-8782-034119aab19c)
![3](https://github.com/user-attachments/assets/1ae606af-ce43-456a-a558-83b8d31ffc78)

**Comment by krystian3w** [comment]
Need to evaluate whether the lock-down heuristics still work for hosts lists and simplified EL syntax.

**Comment by garry-ut99** [comment]
It doesn't work with hosts filters (like `127.0.0.1 shafou.com` from "Dan Pollock’s hosts file" or `shafou.com`) because like already mentioned in uBlock documentation [LINK](https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#hosts-files): 

> All hostname entries from a HOSTS file resource from uBO's point of view will be syntactically equivalent to a filter using the form `||hostname^`.
